gateway_test.go

```diff
@@ -5,11 +5,13 @@ package main
 
```

Comment:
I had to refactor these tests a little because the mock gateway created in `createMockEchoGatewayServer` never actually used `FilteredHttpRequestHandler`, which is where both forbidden/allowed origins and target rewrites are implemented. Now, instead of using `ForbiddenCheckHttpRequestHandler` which would just fire metrics and send a canned response, we wire `MockHttpRequestHandler` into `ProtoHTTPAppHandler` or `BinaryHTTPAppHandler`.